### DISCUSSION TITLE
Stripe integration q

### DISCUSSION THREAD

**Discussion by j2is** [swellstores/community#23] [Help] [answered]
Hey,

I'm using the example from the documentation for Stripe but seem to get a "payment authorised but not yet captured" following a 3DSecure check. I don't believe there's any redirection happening for this (it's not iDeal), just wondering if there is any additional bits I would need to account for?

```
import swell from 'swell-js';

swell.init('my-store', 'pk_...');

swell.payment.createElements({
  card: {
    ...
  },
});

const form = document.getElementById('payment-form');
form.addEventListener('submit', function(event) {
  event.preventDefault();
  showLoading();

  await swell.payment.tokenize({
    card: {
      onError: (err) => {
        // inform the customer there was an error
      },
      onSuccess: () => {
        //finally submit the form
        form.submit();
      }
    }
    // ideal: { onError: (err) => {}, ...}
    // klarna: { onError: (err) => {}, ...}
    // bancontact: { onError: (err) => {}, ...}
  });

  hideLoading();
});
```

**Comment by rodrigoddalmeida** [comment, marked as answer]
HI @j2is, could you check if you have Settings > Orders > Charge automatically enabled?

Also, I noticed that you have added an `await` call to a non-async function in that `submit` handler.

**Reply by j2is** [reply]
@rodrigoddalmeida thanks so much for the reply, charge is auto enabled. I used the example code from swell above (may be update the async function there). In my implementation  I do have an async function. I wondering if the onSuccess handler might not be triggered sometimes after 3D secure

**Reply by rodrigoddalmeida** [reply]
In that case, you can check for the following:

- Confirm if there are any errors sent to the `onError` handler
- Otherwise, check if `onSuccess` is being triggered - if so, `swell.cart.submitOrder()` should be called at some point so the payment authorization can be charged
